src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,30 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login cannot be null");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must be at least 6 characters");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password cannot be null");
+        }
+        if (user.getPassword().length() < 6) {
+            throw new RegistrationException("Password must be at least 6 characters");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age cannot be null");
+        }
+        if (user.getAge() < 18) {
+            throw new RegistrationException("User must be at least 18 years old");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with this login already exists");
+        }
+        return storageDao.add(user);
```

Comment:
Ensure that the `storageDao.add(user)` method is correctly implementing the logic to add a user to the storage. It should not only add the user but also return the correct user object. This is something that should be covered by your unit tests.